ground-core/src/main/java/edu/berkeley/ground/resources/NodesResource.java

```diff
@@ -85,7 +92,9 @@ public NodeVersion createNodeVersion(@Valid NodeVersion nodeVersion, @QueryParam
     @Path("/{name}/latest")
     public List<String> getLatestVersions(@PathParam("name") String name) throws GroundException {
         LOGGER.info("Retrieving the latest version of node " + name + ".");
-        return this.nodeFactory.getLeaves(name);
+        List<String> leaves = this.nodeFactory.getLeaves(name);
+        Set<String> set = new HashSet<>(leaves); //Dedup the leaves
```

Comment:
Does the new query duplicate the leaves? If so, can (should?) we push the deduplication farther down the stack?